stdlib/public/core/Tuple.swift

```diff
@@ -0,0 +1,373 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// Variadic functions for comparison operations on tuple types.
+
+/// Returns a Boolean value indicating whether the corresponding components of
+/// two tuples are equal.
+///
+/// For two tuples to compare as equal, each corresponding pair of components
+/// must be equal. The following example compares tuples made up of 4
+/// components:
+///
+///     let a = ("a", 1, 2, 3)
+///     let b = ("a", 1, 2, 3)
+///     print(a == b)
+///     // Prints "true"
+///
+///     let c = ("a", 1, 2, 4)
+///     print(a == c)
+///     // Prints "false"
+///
+/// - Parameters:
+///   - lhs: A tuple of `Equatable` elements.
+///   - rhs: Another tuple of elements of the same type as `lhs`.
+@_alwaysEmitIntoClient
+public func ==<each B: Equatable>(lhs: (repeat each B), rhs: (repeat each B)) -> Bool {
```

Comment:
aside: does the choice of `B` as the generic parameter name have some particular significance or prior art?

Reply:
Ah, no, this was leftover from originally writing it as `<A, each B>` thinking it might make the algorithm simpler. It didn’t. I’ll rename and make it more verbose!

Reply:
the verbosity budget should be flush, what with all the boilerplate removal! 😄